core-bundle/tests/Controller/FrontendModule/RootPageDependentModulesControllerTest.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Controller\FrontendModule;
+
+use Contao\Controller;
+use Contao\CoreBundle\Cache\EntityCacheTags;
+use Contao\CoreBundle\Controller\FrontendModule\RootPageDependentModulesController;
+use Contao\CoreBundle\Routing\ScopeMatcher;
+use Contao\CoreBundle\Tests\TestCase;
+use Contao\ModuleModel;
+use Contao\PageModel;
+use Contao\System;
+use Contao\Template;
+use Contao\TemplateLoader;
+use PHPUnit\Framework\MockObject\MockObject;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class RootPageDependentModulesControllerTest extends TestCase
+{
+    private ContainerBuilder $container;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->container = $this->getContainerWithContaoConfiguration();
+        $this->container->set('contao.cache.entity_tags', $this->createMock(EntityCacheTags::class));
+
+        System::setContainer($this->container);
+    }
+
+    public function testReturnsWildCardWhenBackendRequest(): void
+    {
+        $scopeMatcher = $this->createMock(ScopeMatcher::class);
+        $scopeMatcher
+            ->expects($this->once())
+            ->method('isBackendRequest')
+            ->willReturn(true)
+        ;
+
+        $router = $this->createMock(RouterInterface::class);
+        $router
+            ->expects($this->once())
+            ->method('generate')
+        ;
+
+        $translator = $this->createMock(TranslatorInterface::class);
+        $translator
+            ->expects($this->once())
+            ->method('trans')
+            ->willReturn('Root Page Dependent Modules')
+        ;
+
+        $container = $this->mockContainer();
+        $container->set('contao.routing.scope_matcher', $scopeMatcher);
+        $container->set('router', $router);
+        $container->set('translator', $translator);
+
+        TemplateLoader::addFile('be_wildcard', __DIR__.'/../../../src/Resources/contao/templates/backend');
```

Comment:
The path should probably be wrapped in `Path:: canonicalize()`.

Reply:
Changed in 41f654ba9f716bf69d92288977a57f50c63ecd3c. 👍